app/src/main/kotlin/com/intive/picover/parties/repository/PartiesRepository.kt

```diff
@@ -20,7 +21,12 @@ class PartiesRepository @Inject constructor(
 			}
 		}
 
-	fun partyById(id: Int) =
+	suspend fun addParty(partyRemote: PartyRemote) =
+		partiesReference.push().key?.let {
+			partiesReference.child(it).setValue(partyRemote.copy(id = it)).await()
```

Comment:
> Firebase clients provide a push() function that generates a unique key for each new child. For more please see: [Firebase Push().key](https://firebase.google.com/docs/database/admin/save-data#section-push)

The idea is to use Firebase's auto-generated key, assign it to the model, and use it to search for it. As the key is a String that is why the model has been changed to String.